src/main/java/com/efficio/fieldbook/web/common/service/impl/CrossingTemplateParser.java

```diff
@@ -149,37 +153,136 @@ private void parseObservationSheet(final String programUUID) throws FileParsingE
+	 * @param femaleStudyName - name of female study
+	 * @param femalePlotNos - plot numbers to look up for female study
+	 * @param maleStudiesWithPlotNos - map of male studies to corresponsing male plot numbers
+	 * @param entryIdToCrossInfoMap - map of entry id to study name
+	 * @throws FileParsingException
+	 */
+
+	void lookupCrossParents(final String femaleStudyName, final Set<Integer> femalePlotNos,
+		final Map<String, Set<Integer>> maleStudiesWithPlotNos, final Map<Integer, Triple<String, Integer, List<Integer>>> entryIdToCrossInfoMap) throws FileParsingException {
+
+		final Map<Integer, StudyGermplasmDto> femalePlotMap =
+			this.getPlotNoToStudyGermplasmDtoMapForStudy(femaleStudyName, femalePlotNos);
+		// Create map of male studies to its plotToStudyGermplasmDtoMap lookup
+		// for each male study, lookup the associated StudyGermplasmDto of specified male plot #s
+		final Map<String, Map<Integer, StudyGermplasmDto>> maleNurseriesPlotMap = new HashMap<>();
+		for (final Map.Entry<String, Set<Integer>> entry : maleStudiesWithPlotNos.entrySet()) {
+			final String maleStudyName = entry.getKey();
+			maleNurseriesPlotMap
+				.put(maleStudyName, this.getPlotNoToStudyGermplasmDtoMapForStudy(maleStudyName, entry.getValue()));
+		}
+
+		// Set looked up GIDs and names of parents to ImportedCrosses object
+		for (final ImportedCrosses cross : this.importedCrossesList.getImportedCrosses()) {
+			final Triple<String, Integer, List<Integer>> crossInfo = entryIdToCrossInfoMap.get(cross.getEntryId());
+			if (femalePlotMap.containsKey(crossInfo.getMiddle())) {
+				final StudyGermplasmDto femaleCrossParent = femalePlotMap.get(crossInfo.getMiddle());
+				final ImportedGermplasmParent femaleParent = new ImportedGermplasmParent(femaleCrossParent.getGermplasmId(), femaleCrossParent.getDesignation(), crossInfo.getMiddle(),
+					femaleStudyName);
+				cross.setFemaleParent(femaleParent);
+			} else {
+				throw new FileParsingException(this.messageSource.getMessage("no.list.data.for.plot",
+					new Object[] {femaleStudyName, crossInfo.getMiddle()}, LocaleContextHolder.getLocale()));
+			}
+
+			cross.setMaleParents(this.getMaleParents(maleNurseriesPlotMap, crossInfo));
+
 			final Germplasm germplasm = new Germplasm();
 			germplasm.setGnpgs(2);
 			germplasm.setGid(Integer.MAX_VALUE);
-			germplasm.setGpid1(femaleListDataProject.getGermplasmId());
-			germplasm.setGpid2(maleListDataProjects.get(0).getGermplasmId());
-			final String crossString = this.crossingService.getCross(germplasm, importedCrosses, "/");
-			importedCrosses.setCross(crossString);
+			germplasm.setGpid1(Integer.valueOf(cross.getFemaleGid()));
+			germplasm.setGpid2(cross.getMaleParents().get(0).getGid());
+			final String crossString = this.crossingService.getCross(germplasm, cross, "/");
+			cross.setCross(crossString);
+		}
+	}
 
-			this.importedCrossesList.addImportedCrosses(importedCrosses);
+	List<ImportedGermplasmParent> getMaleParents(
```

Comment:
getMaleParents name was a bit confusing to me given that it is a cross attribute and in fact is loading the male parents. Probably we should use a more appropriated name @darla-leafnode

Reply:
A cross here can have one or multiple male parents (polycross) as opposed to just having only one female parent at all time)